ssl/start.sh

```diff
@@ -1,7 +1,46 @@
 #!/bin/sh
 
-if [ -f /kopano/ssl/ca.pem ]; then
-	exit 0
+mkdir -p /kopano/ssl/clients/
+
+if [ ! -f /kopano/ssl/ca.pem ]; then
+	# https://github.com/google/easypki
+	echo "Creating CA and server certificates..."
+	easypki create --filename internalca --organizational-unit primary --expire 3650 --ca "Internal Kopano System"
+
+	for s in kserver kdagent kmonitor ksearch kspooler kwebapp; do
+		easypki create --ca-name internalca --organizational-unit $s --expire 3650 $s
+		cp /kopano/easypki/internalca/keys/$s.key /tmp/$s.pem
+		cat /kopano/easypki/internalca/certs/$s.crt >> /tmp/$s.pem
+		openssl x509 -in /kopano/easypki/internalca/certs/$s.crt -pubkey -noout > /tmp/$s-public.pem
+		cp /tmp/$s.pem /kopano/ssl/$s.pem
+		cp /tmp/$s-public.pem /kopano/ssl/clients/$s-public.pem
```

Comment:
cp is not atomic while rm has the chance to be atomic. So also the choice of location matters. Do not use tmp, use a tmp file in the same folder as the destination to write to and after having created the file, move it to the desired name in the same folder. So no /tmp, no cp.

Reply:
This is for all the create stuff also in other files and down in this script. (Since mv is not atomic when crossing file systems and /tmp is most certainly a different file system like tmpfs).

Reply:
ok, misconception from my side. not files are created in their location with a tmp file extension and moved afterwards.